src/test/java/onlinebookstore/service/BookServiceTest.java

```diff
@@ -0,0 +1,193 @@
+package onlinebookstore.service;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+
+import java.math.BigDecimal;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import onlinebookstore.dto.book.BookDto;
+import onlinebookstore.dto.book.BookDtoWithoutCategoryIds;
+import onlinebookstore.dto.book.CreateBookRequestDto;
+import onlinebookstore.exception.EntityNotFoundException;
+import onlinebookstore.mapper.BookMapper;
+import onlinebookstore.model.Book;
+import onlinebookstore.repository.BookRepository;
+import onlinebookstore.service.impl.BookServiceImpl;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class BookServiceTest {
+    @Mock
+    private BookMapper bookMapper;
+
+    @Mock
+    private BookRepository bookRepository;
+
+    @InjectMocks
+    private BookServiceImpl bookService;
+
+    @Test
+    @DisplayName("Save valid book")
+    void saveBook_validRequest_returnBookDto() {
+        Long bookId = 1L;
+        Book book = new Book();
+        book.setId(bookId);
+        book.setTitle("Clean Code");
+        book.setAuthor("Robert Martin");
+        book.setIsbn("00-000-000-01");
+        book.setPrice(new BigDecimal("100.00"));
+        book.setDescription("some description");
+        book.setCoverImage("https://some-image-url.com");
+        book.setCategories(new HashSet<>());
```

Comment:
if you use the same test object better move it to the private method and avoid duplication.